### PR TITLE
fixed azurerm_spatial_anchors_account tags argument always forces replacement of spatial_anchors_account resource #11037

### LINKED ISSUE #11037: azurerm_spatial_anchors_account tags argument always forces replacement of spatial_anchors_account resource
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
```
Terraform v0.14.8
+ provider registry.terraform.io/hashicorp/azurerm v2.51.0
```

### Affected Resource(s)

* `azurerm_spatial_anchors_account`

### Terraform Configuration Files

**main.tf:**
```hcl
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = ">= 2.48"
    }
  }

provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "resource_group" {
  name     = "rg-asa-example"
  location = "West Europe"
}

resource "azurerm_spatial_anchors_account" "spatial_anchors_account" {
  name                = "asa-example-101"
  resource_group_name = azurerm_resource_group.resource_group.name
  location            = azurerm_resource_group.resource_group.location
  tags                = {  
    "foo" = "bar" }
}
```

### Expected Behaviour
Terraform applies change to `azurerm_spatial_anchors_account.spatial_anchors_account` **without force replacement of resource** when tags argument is changed.
```hcl
# azurerm_spatial_anchors_account.spatial_anchors_account must be replaced
-/+ resource "azurerm_spatial_anchors_account" "spatial_anchors_account" {
      ~ id                  = "/subscriptions/f4b25ef4-e1a9-468c-b855-7417a32e0bf1/resourceGroups/rg-asa-example/providers/Microsoft.MixedReality/spatialAnchorsAccounts/asa-example-101" -> (known after apply)
        name                = "asa-example-101"
      ~ tags                 = {
          ~ "foo" = "bar" -> "foo"
        }
        # (2 unchanged attributes hidden)
    }
```

### Actual Behaviour
Terraform applies removal and re-creation of `azurerm_spatial_anchors_account.spatial_anchors_account` **with force replacement of resource** when tags argument is changed.
```hcl
# azurerm_spatial_anchors_account.spatial_anchors_account must be replaced
-/+ resource "azurerm_spatial_anchors_account" "spatial_anchors_account" {
      ~ id                  = "/subscriptions/f4b25ef4-e1a9-468c-b855-7417a32e0bf1/resourceGroups/rg-asa-example/providers/Microsoft.MixedReality/spatialAnchorsAccounts/asa-example-101" -> (known after apply)
        name                = "asa-example-101"
      ~ tags                = { # forces replacement
          ~ "foo" = "bar" -> "foo"
        }
        # (2 unchanged attributes hidden)
    }
```

### Steps to Reproduce

1. `terraform apply` to create resource `azurerm_spatial_anchors_account.spatial_anchors_account`
2. Modify k/v-pair in tags argument of  resource `azurerm_spatial_anchors_account.spatial_anchors_account`
2. `terraform apply or terraform plan`

### References

* #0000

### PR DESCRIPTION
**Pull request by kohei3110**
### Acceptance Test
```
$ make acctests SERVICE='mixedreality' TESTTIMEOUT='60m'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./azurerm/internal/services/mixedreality  -timeout 60m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
2021/05/27 21:54:58 [DEBUG] not using binary driver name, it's no longer needed
2021/05/27 21:54:58 [DEBUG] not using binary driver name, it's no longer needed
=== RUN   TestAccSpatialAnchorsAccount_basic
=== PAUSE TestAccSpatialAnchorsAccount_basic
=== RUN   TestAccSpatialAnchorsAccount_complete
=== PAUSE TestAccSpatialAnchorsAccount_complete
=== CONT  TestAccSpatialAnchorsAccount_basic
=== CONT  TestAccSpatialAnchorsAccount_complete
--- PASS: TestAccSpatialAnchorsAccount_complete (117.44s)
--- PASS: TestAccSpatialAnchorsAccount_basic (119.50s)
PASS
ok  	github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/mixedreality	120.035s
Kohei:terraform-provider-azurerm kohei$ 
Kohei:terraform-provider-azurerm kohei$ 
Kohei:terraform-provider-azurerm kohei$ make acctests SERVICE='mixedreality' TESTTIMEOUT='60m'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./azurerm/internal/services/mixedreality  -timeout 60m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
2021/05/27 22:40:58 [DEBUG] not using binary driver name, it's no longer needed
2021/05/27 22:40:58 [DEBUG] not using binary driver name, it's no longer needed
=== RUN   TestAccSpatialAnchorsAccount_basic
=== PAUSE TestAccSpatialAnchorsAccount_basic
=== RUN   TestAccSpatialAnchorsAccount_complete
=== PAUSE TestAccSpatialAnchorsAccount_complete
=== CONT  TestAccSpatialAnchorsAccount_basic
=== CONT  TestAccSpatialAnchorsAccount_complete
2021/05/27 22:41:32 [DEBUG] AzureRM Response for https://management.azure.com/subscriptions/5464d977-f14b-44b9-8da2-b33250ef5393/resourceGroups/acctestRG-mr-210527224058704068/providers/Microsoft.MixedReality/spatialAnchorsAccounts/acCTestdf210527224058704068?api-version=2019-02-28-preview: 
HTTP/2.0 200 OK
Cache-Control: no-cache
Content-Type: application/json; charset=utf-8
Date: Thu, 27 May 2021 13:41:32 GMT
Expires: -1
Ms-Cv: ydYj1y5bBU+2GTcNerRyDg.0
Pragma: no-cache
Strict-Transport-Security: max-age=31536000; includeSubDomains
Vary: Accept-Encoding
X-Content-Type-Options: nosniff
X-Ms-Correlation-Request-Id: 4de663f8-a640-cb08-7e32-4cb992a520ec
X-Ms-Ratelimit-Remaining-Subscription-Reads: 11988
X-Ms-Request-Id: c711e371-6947-475a-b6b7-276e7f7ae024
X-Ms-Routing-Request-Id: JAPANEAST:20210527T134132Z:c711e371-6947-475a-b6b7-276e7f7ae024

{"id":"/subscriptions/5464d977-f14b-44b9-8da2-b33250ef5393/resourceGroups/acctestRG-mr-210527224058704068/providers/Microsoft.MixedReality/spatialAnchorsAccounts/acCTestdf210527224058704068","name":"acCTestdf210527224058704068","type":"Microsoft.MixedReality/spatialAnchorsAccounts","identity":null,"location":"eastus","plan":null,"sku":null,"kind":null,"tags":{"Environment":"Production"},"properties":{"accountDomain":"eastus.mixedreality.azure.com","accountId":"73be8f26-3641-46ca-b58c-229f7186114c"}}
2021/05/27 22:41:32 [DEBUG] TF_ACCTEST_REATTACH set to 1 and acctest.TestHelper is not nil, using reattach-based testing
--- PASS: TestAccSpatialAnchorsAccount_basic (116.12s)
--- PASS: TestAccSpatialAnchorsAccount_complete (116.59s)
PASS
ok  	github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/mixedreality	117.217s
```

#### main.tf
```
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = ">= 2.48"
    }
  }
}

provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "resource_group" {
  name     = "rg-asa-example"
  location = "West Europe"
}

resource "azurerm_spatial_anchors_account" "spatial_anchors_account" {
  name                = "asa-example-101"
  resource_group_name = azurerm_resource_group.resource_group.name
  location            = azurerm_resource_group.resource_group.location
  tags                = {  
    "foo" = "fuga" }
}
```

#### tf plan & tf apply
```
$ terraform plan -out=dev.out

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # azurerm_resource_group.resource_group will be created
  + resource "azurerm_resource_group" "resource_group" {
      + id       = (known after apply)
      + location = "westeurope"
      + name     = "rg-asa-example"
    }

  # azurerm_spatial_anchors_account.spatial_anchors_account will be created
  + resource "azurerm_spatial_anchors_account" "spatial_anchors_account" {
      + id                  = (known after apply)
      + location            = "westeurope"
      + name                = "asa-example-101"
      + resource_group_name = "rg-asa-example"
      + tags                = {
          + "foo" = "hoge"
        }
    }

Plan: 2 to add, 0 to change, 0 to destroy.

------------------------------------------------------------------------

This plan was saved to: dev.out

To perform exactly these actions, run the following command to apply:
    terraform apply "dev.out"

Kohei:tf kohei$ terraform apply "dev.out"

Warning: Provider development overrides are in effect

The following provider development overrides are set in the CLI configuration:
 - hashicorp/azurerm in /Users/Kohei/go/bin/bin

The behavior may therefore not match any released version of the provider and
applying changes may cause the state to become incompatible with published
releases.

azurerm_resource_group.resource_group: Creating...
azurerm_resource_group.resource_group: Creation complete after 3s [id=/subscriptions/5464d977-f14b-44b9-8da2-b33250ef5393/resourceGroups/rg-asa-example]
azurerm_spatial_anchors_account.spatial_anchors_account: Creating...
azurerm_spatial_anchors_account.spatial_anchors_account: Still creating... [10s elapsed]
azurerm_spatial_anchors_account.spatial_anchors_account: Creation complete after 12s [id=/subscriptions/5464d977-f14b-44b9-8da2-b33250ef5393/resourceGroups/rg-asa-example/providers/Microsoft.MixedReality/spatialAnchorsAccounts/asa-example-101]

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

The state of your infrastructure has been saved to the path
below. This state is required to modify and destroy your
infrastructure, so keep it safe. To inspect the complete state
use the `terraform show` command.

```

### Modify a Tag value
```
$ terraform plan -out=dev.out
azurerm_resource_group.resource_group: Refreshing state... [id=/subscriptions/5464d977-f14b-44b9-8da2-b33250ef5393/resourceGroups/rg-asa-example]
azurerm_spatial_anchors_account.spatial_anchors_account: Refreshing state... [id=/subscriptions/5464d977-f14b-44b9-8da2-b33250ef5393/resourceGroups/rg-asa-example/providers/Microsoft.MixedReality/spatialAnchorsAccounts/asa-example-101]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # azurerm_spatial_anchors_account.spatial_anchors_account will be updated in-place
  ~ resource "azurerm_spatial_anchors_account" "spatial_anchors_account" {
        id                  = "/subscriptions/5464d977-f14b-44b9-8da2-b33250ef5393/resourceGroups/rg-asa-example/providers/Microsoft.MixedReality/spatialAnchorsAccounts/asa-example-101"
        name                = "asa-example-101"
      ~ tags                = {
          ~ "foo" = "hoge" -> "fuga"
        }
        # (2 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

------------------------------------------------------------------------

This plan was saved to: dev.out

To perform exactly these actions, run the following command to apply:
    terraform apply "dev.out"
```

(fixes #11037)

### CODE REVIEW

File: azurerm/internal/services/mixedreality/spatial_anchors_account_resource.go [hunk 1]
```diff
@@ -20,21 +20,20 @@ import (
...
-		Create: resourceSpatialAnchorsAccountCreate,
+		Create: resourceSpatialAnchorsAccountCreateUpdate,
 		Read:   resourceSpatialAnchorsAccountRead,
+		Update: resourceSpatialAnchorsAccountCreateUpdate,
 		Delete: resourceSpatialAnchorsAccountDelete,
 		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
 			_, err := parse.SpatialAnchorsAccountID(id)
 			return err
 		}),
-
```
Comment (katbyte) [inline azurerm/internal/services/mixedreality/spatial_anchors_account_resource.go:30]: could we revert all this white space removal? thanks

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @kohei3110 - looks good but can we add all that whitespace back in? as well as update the docs that this no longer forces a new resoruce

**Comment by kohei3110** [comment]
Plus, since there was no such description as tag changes would force a new resource, I confirmed that no modification in the document was necessary.

**Review by katbyte — APPROVED** [review]
Thanks @kohei3110 LGTM! 👍

**Comment by katbyte** [comment]
just needs conflicts resolves and this this can be merged

**Pull request merged**